interpreter/utils/edit_code.py

```diff
@@ -0,0 +1,64 @@
+import os
+from yaspin import yaspin
+
+from .temporary_file import create_temporary_file, cleanup_temporary_file
+from .open_file import open_file
+from ..code_interpreters.languages.utils.language_tools import get_language_file_extension, get_language_proper_name
+
+
+def edit_code(code, language, interpreter):
+    """
+    Edit the code and listen for changes with watchdog
+    """
+
+    temp_code = code
+
+    temp_file = create_temporary_file(
+        temp_code, get_language_file_extension(language), verbose=interpreter.debug_mode
+    )
+
+    language_name = get_language_proper_name(language)
+
+    file_name = os.path.basename(temp_file)
+
+    if interpreter.debug_mode:
+        print(f"Editing {language_name} code in {file_name}")
+        print("---")
+
+    # Run semgrep
+    try:
+        print("  Press `ENTER` after you've saved your edits.")
+
+        open_file(temp_file)
```

Comment:
Ideally, eventually we'll be able to incorporate some sort of in-place editing that feels nice, but for the proof of concept, it seemed more prudent to leverage the user's existing default editor.